### PR TITLE
add resource_template configuration resource

### PR DESCRIPTION
**Pull request by Hedgehog57**
**Description:**

**Pull request in home-assistant (if applicable):** home-assistant/home-assistant#30703

## Checklist:

- [ ] Branch: `next` is for changes and new documentation that will go public with the next Home Assistant release. Fixes, changes and adjustments for the current release should be created against `current`.
- [ ] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

**Comment by klaasnicolaas** [comment]
Please rebase this PR to the `next` branch

**Comment by frenck** [comment]
Rebased the PR onto the `next` branch. Please @Hedgehog57, make sure you use the correct branches in the future. 👍

File: source/_integrations/binary_sensor.rest.markdown [hunk 1]
```diff
@@ -49,13 +49,25 @@ binary_sensor:
...
 
 {% configuration %}
 resource:
   description: The resource or endpoint that contains the value.
   required: true
   type: string
   default: string
+resource_template:
+  description: The resource or endpoint that contains the value with template support.
+  required: true
```
Comment (frenck) [inline source/_integrations/binary_sensor.rest.markdown]: It is not required according to the codebase.

**Review by frenck — CHANGES_REQUESTED** [review]
Hi there, @Hedgehog57!

I've left you a comment on this PR, could you please take a look? Thanks! 👍

**Comment by Hedgehog57** [comment]
@frenck thanks for Your intervention. I'm not a real developer so I make mistakes.
Required was because entity must have one of two - resource or resource_template. If there is resource_template you can skip resource. But I don't know how it must be written.

File: source/_integrations/binary_sensor.rest.markdown (hunk 1 above)
Comment (Hedgehog57) [reply source/_integrations/binary_sensor.rest.markdown]: changed to required: false

**Review by frenck — APPROVED** [review]
No problem sir! I understand we all have to learn at some point, and I respect you for doing that 👍 

✅ Approved. Can be merged as soon as the parent PR gets merged.

**Review by balloob — APPROVED** [review]

**Pull request merged**